frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelEnvironmentSwitches/FeatureOverviewSidePanelEnvironmentSwitches.tsx

```diff
@@ -89,6 +102,34 @@ export const FeatureOverviewSidePanelEnvironmentSwitches = ({
                             <StyledSubLabel>
                                 {strategiesLabel}
                                 {variantsLink}
+                                <ConditionallyRender
+                                    condition={
```

Comment:
I'd extract the condition to the intention-revealing name

Reply:
Not sure if I can make it better than:
```
environment.enabled &&
variants.length == 0 &&
someEnabledEnvironmentHasVariants
```
If you're suggesting to extract this in a single variable/method it would be something like: `isEnabledWithoutVariantsAndOtherEnabledEnvironmentsHaveVariants(environment)`, in which case I prefer my version, but if you're suggesting something different please let me know (I can also change it after merging)

Reply:
maybe showVariantWarning: boolean?

Reply:
I decided to use `hasWarning` because there was also the idea of using the same warning for other purposes in the future.